### ISSUE TITLE
input size

### ISSUE THREAD

**Issue by Lelantos42** [karolzak/keras-unet#40]
I seem to be having issues with the input size for the vanilla_unet.  I am wanting to test this with (64,64,1) or (64,64,3) images, but I receive this error:

ValueError: Negative dimension size caused by subtracting 3 from 2 for '{{node conv2d_7/Conv2D}} = Conv2D[T=DT_FLOAT, data_format="NHWC", dilations=[1, 1, 1, 1], explicit_paddings=[], padding="VALID", strides=[1, 1, 1, 1], use_cudnn_on_gpu=true](conv2d_6/Relu, conv2d_7/Conv2D/ReadVariableOp)' with input shapes: [?,2,2,64], [3,3,64,64].

**Comment by karolzak** [comment]
Hi @Lelantos42 
Seems like your input size is too small.
`vanilla_unet` is an implementation of the [original paper](https://arxiv.org/pdf/1505.04597.pdf) and same as in the original paper it's using `valid` padding method for all the conv layers, which means 2 px are substracted at each conv layer like in the schema bellow:
![image](https://user-images.githubusercontent.com/9843623/96283655-ceb32200-0fdc-11eb-836d-239ca81909e2.png)
At each pooling layer you're also downsizing input to 1/2 of size so for 572x572 input (like in the paper) you end up with features map vector of 28x28x1024 at the lowest level.
So in your case what is happening is that somewhere in the bottom left area of that U-shape architecture your features map size is simply too small to run convolutions on it.
Long story short - you need bigger images. I think 256x256 should be safe (they used 572x572 in the original paper) also good to note that original images were a bit smaller and they applied mirroring on the edge/border to enlarge them and thus compensate for not using padding in convolution layers. Result of that is that while input image is 572x572 the output map is just 388x388 (due to all the pixels lost by not using padding in convolutions). 
It's a very interesting paper btw - strongly advise to read it :)

**Comment by Lelantos42** [comment]
This was very helpful! :) I will definitely give it a read. Thank you!
